File: editorial/chap04/answer_B16.py
# 入力
N = int(input())
H = list(map(int, input().split()))

# 配列の準備
dp = [ None ] * N

# 動的計画法
dp[0] = 0
dp[1] = abs(H[0] - H[1])
for i in range(2, N):
	dp[i] = min(dp[i-1] + abs(H[i-1]-H[i]), dp[i-2] + abs(H[i-2]-H[i]))

# 出力
print(dp[N-1])

File: editorial/chap04/cpp/answer_B16.cpp
#include <iostream>
#include <cmath>
#include <algorithm>
using namespace std;

int N, H[100009];
int dp[100009];

int main() {
	// 入力
	cin >> N;
	for (int i = 1; i <= N; i++) cin >> H[i];

	// 動的計画法
	dp[1] = 0;
	dp[2] = abs(H[1] - H[2]);
	for (int i = 3; i <= N; i++) {
		dp[i] = min(dp[i - 1] + abs(H[i - 1] - H[i]), dp[i - 2] + abs(H[i - 2] - H[i]));
	}

	// 出力
	cout << dp[N] << endl;
	return 0;
}
